sys/net/gnrc/transport_layer/tcp/gnrc_tcp_fsm.c

```diff
@@ -0,0 +1,716 @@
+
+            *notify_owner = true;
+            break;
+
+        case LISTEN:
+            /* Clear Adress Info */
+            if(conn->status & GNRC_TCP_STATUS_ALLOW_ANY_ADDR){
+#ifdef MODULE_GNRC_IPV6
+                ipv6_addr_set_unspecified((ipv6_addr_t *) &(conn->local_addr));
+#endif
+            }
+#ifdef MODULE_GNRC_IPV6
+            ipv6_addr_set_unspecified((ipv6_addr_t *) &(conn->peer_addr));
+#endif
+            conn->peer_port = GNRC_TCP_PORT_UNSPEC;
+
+            /* Allocate rcv Buffer */
+            if (conn->rcv_buf_raw == NULL) {
+                conn->rcv_buf_raw = _rcvbuf_alloc();
+                if (conn->rcv_buf_raw == NULL) {
+                    DEBUG("gnrc_tcp_fsm : _translate_to() : Can't allocate rcv_buf_raw\n");
+                    return -ENOMEM;
+                }
+                else {
+                    ringbuffer_init(&conn->rcv_buf, conn->rcv_buf_raw, GNRC_TCP_RCV_BUF_SIZE);
+                }
+            }
+
+            /* Add to Connection to active connection (if not already active) */
+            mutex_lock(&_list_conn_tcp_lock);
+            LL_FOREACH(_list_conn_tcp_head, iter) {
+                if (iter == conn) {
+                    found = 1;
+                }
+            }
+            if (!found) {
+                LL_APPEND(_list_conn_tcp_head, conn);
+            }
+            mutex_unlock(&_list_conn_tcp_lock);
+
+            /* Clear Accepted Flag */
+            conn->status &= ~GNRC_TCP_STATUS_ACCEPTED;
+            break;
+
+        case SYN_SENT:
+            /* Allocate rcv Buffer */
+            if (conn->rcv_buf_raw == NULL) {
+                conn->rcv_buf_raw = _rcvbuf_alloc();
+                if (conn->rcv_buf_raw == NULL) {
+                    DEBUG("gnrc_tcp_fsm : _translate_to() : Can't allocate rcv_buf_raw\n");
+                    return -ENOMEM;
+                }
+                else {
+                    ringbuffer_init(&conn->rcv_buf, conn->rcv_buf_raw, GNRC_TCP_RCV_BUF_SIZE);
+                }
+            }
+
+            /* Add to Connections to active connection (if not already active) */
+            mutex_lock(&_list_conn_tcp_lock);
+            LL_FOREACH(_list_conn_tcp_head, iter) {
+                if (iter == conn) {
+                    found = 1;
+                }
+            }
+            if (!found) {
+                LL_APPEND(_list_conn_tcp_head, conn);
+            }
+            mutex_unlock(&_list_conn_tcp_lock);
+
+            /* Clear Accepted Flag */
+            conn->status &= ~GNRC_TCP_STATUS_ACCEPTED;
+            break;
+
+        case ESTABLISHED:   *notify_owner = true;
+            break;
+
+        case CLOSE_WAIT:    *notify_owner = true;
+            break;
+
+        case TIME_WAIT:     _restart_timewait_timer(conn);
+            break;
+
+        default:            break;
+    }
+    conn->state = state;
+    return 0;
+}
+
+/**
+ * @brief real fsm: needs to be protected from the outside
```

Comment:
Note that doxygen in c-files will not be included into doxygen build, so you can be a little bit more relaxed here ;-)